app/controllers/comment_controller.php

```diff
@@ -55,4 +62,93 @@ public function write()
         $this->set(get_defined_vars());
         $this->render($page);
     }
+
+    public function edit_comment()
+    {
+        check_user_session();
+        $comment = new Comment();
+        $comment_id = get_comment_id_from_url();
+        
+        $comment->id = $comment_id;
+
+        $thread_id = get_thread_id_from_url();
+        $thread = Thread::get($thread_id);
+        $comment = $comment->get($comment_id);
```

Comment:
You can simplify this by making this method static. With this, you won't need to instantiate a new object like in L69, assign id on that object in L72, then finally overwrite the same object like you did here when you can simply do something like this
```php
$comment = Comment::get(Param::get('comment_id'));
```

in model : 
```php
public static function get($id)
{
    $db = DB::conn();
    $row = $db->row('SELECT * FROM comment WHERE id = ?', array($id));
    
    if (!$row) {
        throw new RecordNotFoundException('...');
    }

    return new self($row);
}
```